scripts/scil_bundle_mean_fixel_mrds_metric.py

```diff
@@ -0,0 +1,125 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+
+"""
+Given a bundle and MRDS metrics, compute the fixel-specific
+metrics at each voxel intersected by the bundle. Intersected voxels are
+found by computing the intersection between the voxel grid and each streamline
+in the input tractogram.
+
+This script behaves like scil_bundle_mean_fixel_afd.py for fODFs,
+but here for MRDS metrics. These latest distributions add the unique
+possibility to capture fixel-based fractional anisotropy (fixel-FA), mean
+diffusivity (fixel-MD), radial diffusivity (fixel-RD) and
+axial diffusivity (fixel-AD).
+
+Fixel-specific metrics are metrics extracted from Bingham distributions fitted
+to fODF. Their are as many values per voxel as there are lobes extracted. The
+values chosen for a given voxel is the one belonging to the lobe better aligned
+with the current streamline segment.
+
+Output metrics will be name: [prefix]_mrds_f[metric_name].nii.gz
+
+Please use a bundle file rather than a whole tractogram.
+"""
+
+import argparse
+
+import nibabel as nib
+import numpy as np
+
+from scilpy.io.streamlines import load_tractogram_with_reference
+from scilpy.io.utils import (add_overwrite_arg,
+                             add_reference_arg,
+                             assert_headers_compatible,
+                             assert_inputs_exist, assert_outputs_exist)
+from scilpy.tractanalysis.mrds_along_streamlines \
+    import mrds_metrics_along_streamlines
+
+
+def _build_arg_parser():
+    p = argparse.ArgumentParser(description=__doc__,
+                                formatter_class=argparse.RawTextHelpFormatter)
+    p.add_argument('in_bundle',
+                   help='Path of the bundle file.')
+    p.add_argument('in_pdds',
+                   help='Path of the MRDS PDDs volume.')
+
+    g = p.add_argument_group(title='MRDS metrics input')
+    g.add_argument('--fa',
```

Comment:
I am not sure, but I feel like this script could be general to any other metric that we invent some day, as long as it has the right shape.

I would just add one required arg, `(--input_metrics, required=True, action='append'`), and you loop on each input. Then you ask the user for the output filenames and make sure there are as many as the number of inputs.

No?

Reply:
These metrics are DTI specific metrics, and also specific to MRDS solutions. I would keep it that way for now.